### ISSUE TITLE
Opening an image, then opening from another directory yields "HUP detected, closing client" and freeze

### ISSUE THREAD

**Issue by jwhendy** [BestImageViewer/geeqie#867]
### SUMMARY
I started having issues with geeqie a few days and tracked it to [this arch bug](https://bugs.archlinux.org/task/69569), which points to [this glib issue](https://gitlab.gnome.org/GNOME/glib/-/issues/2323).

They declared it fixed with `glib2-2.66.6-2` (which I'm on), but geeqie is still not functioning properly.

The behavior is that geeqie will open once and operate directoy within a directory, but if I try to open a second file from the file manager (thunar in my case) with in that same directory or another, geeqie becomes unresponsive and non-recoverable. I have to `killall geeqie` and start over.

### STEPS TO REPRODUCE

I can do this a few ways:

-  open `dir1/img1`, then try to open `dir1/img2` or `dir2/img1` from thunar
- `xdg-open dir1/img1`, then open another terminal and do `xdg-open some-other-image`
- as pointed out in the arch linux issue linked above, `xdg-open` triggers the `-r` option. Running that directly gives the following output when I try to open the second image, and I'm not sure how to pinpoint further:
```
$ geeqie --debug -r
debugging output enabled (level 1)
    0.000133 (+00000.000133) main: gtk_init
    0.090535 (+00000.090402) main: pixbuf_inline_register_stock_icons
    0.090925 (+00000.000390) main: setting default options before commandline handling
Home directory: /home/jwhendy
Home directory: /home/jwhendy
Home directory: /home/jwhendy
    0.090971 (+00000.000046) main: parse_command_line
    0.091019 (+00000.000048) main: mkdir_if_not_exists
Home directory: /home/jwhendy
    0.091288 (+00000.000269) main: load_options
loader reported [wmf] [Windows Metafile] [.wmf;.apm]
loader reported [ani] [Windows animated cursor] [.ani]
loader reported [avif] [AV1 Image File Format] [.avif]
loader reported [bmp] [BMP] [.bmp]
loader reported [gif] [GIF] [.gif]
loader reported [heif/avif] [HEIF/AVIF Image] [.heif;.heic;.avif]
loader reported [icns] [MacOS X icon] [.icns]
loader reported [ico] [Windows icon] [.ico;.cur]
loader reported [jpeg] [JPEG] [.jpeg;.jpe;.jpg]
loader reported [png] [PNG] [.png]
loader reported [pnm] [PNM/PBM/PGM/PPM] [.pnm;.pbm;.pgm;.ppm]
loader reported [qtif] [QuickTime] [.qtif;.qif]
loader reported [svg] [Scalable Vector Graphics] [.svg;.svgz;.svg.gz]
loader reported [tga] [Targa] [.tga;.targa]
loader reported [tiff] [TIFF] [.tiff;.tif]
loader reported [xbm] [XBM] [.xbm]
loader reported [xpm] [XPM] [.xpm]
    0.092199 (+00000.000911) layout_new: start
    0.092874 (+00000.000675) layout_actions_setup: start
    0.105428 (+00000.012554) layout_actions_setup: add menu
    0.105633 (+00000.000205) layout_actions_setup: add toolbar
    0.105655 (+00000.000022) layout_actions_setup: marks
    0.106696 (+00000.001041) layout_actions_setup: editors
    0.106720 (+00000.000024) layout_actions_setup: status_update_write
    0.106727 (+00000.000007) layout_actions_setup: actions_add_window
    0.106734 (+00000.000007) layout_actions_setup: end
Setting stereo mode 0000 for imd 0x555d98a2a3a0
update size: 0x555d98ecc110  0 0   0 0
    0.152103 (+00000.045369) vflist_refresh: read dir
    0.152119 (+00000.000016) vflist_refresh: populate view
    0.152125 (+00000.000006) vflist_refresh: free filelist
    0.152128 (+00000.000003) vflist_refresh: done
    0.168983 (+00000.016855) pixbuf renderer updated - started drawing 0x555d98ea0210, img: 200x70
    0.168997 (+00000.000014) pixbuf renderer updated - started drawing 0x555d98ea0210, img: 200x70
    0.169003 (+00000.000006) pixbuf renderer updated - started drawing 0x555d98ea0210, img: 200x70
update size: 0x555d98ecc110  0 0   217 964
    0.186446 (+00000.017443) pixbuf renderer updated - started drawing 0x555d98ea0210, img: 200x70
    0.192148 (+00000.005702) layout_new: end
Register realtime 0 /home/jwhendy
Too frequent update of /
Too frequent update of /home
    0.199865 (+00000.007717) vflist_refresh: read dir
    0.201138 (+00000.001273) vflist_refresh: sort
    0.201149 (+00000.000011) vflist_refresh: populate view
    0.201345 (+00000.000196) vflist_refresh: free filelist
    0.201353 (+00000.000008) vflist_refresh: done
    0.201382 (+00000.000029) image reset
    0.201395 (+00000.000013) image begin
new image loader 0x555d98791c10, bufsize=4096 idle_loop=1
exif read /home/jwhendy/test.png, sidecar: -
Thread pool num threads: 1
Register realtime 0 /home/jwhendy/test.png
    0.202873 (+00000.001478) read ahead cancelled for :null
read ahead set to :/home/jwhendy/grab_2020-04-12_123636.png
added Xmp.dc.subject /home/jwhendy/test.png

    0.240263 (+00000.037390) pixbuf renderer updated - started drawing 0x555d98ea0210, img: 200x70
Loading options from /home/jwhendy/.config/geeqie/geeqierc.xml ... done
CLUTTER_FEATURE_SWAP_EVENTS 0
    0.241933 (+00000.001670) layout_editors_reload_start
    0.242519 (+00000.000586) main: gtk_main
image_load_size_cb: 618x717
    0.242960 (+00000.000441) pixbuf renderer updated - started drawing 0x555d98ea0210, img: 618x717
    0.242971 (+00000.000011) pixbuf renderer updated - started drawing 0x555d98ea0210, img: 618x717
image activate focus_in 0
    0.272265 (+00000.029294) image done
Notify pane_histogram: /home/jwhendy/test.png 0004
    0.272302 (+00000.000037) read ahead started for :/home/jwhendy/grab_2020-04-12_123636.png
new image loader 0x555d992d42b0, bufsize=4096 idle_loop=1
exif read /home/jwhendy/grab_2020-04-12_123636.png, sidecar: -
Thread pool num threads: 1
freeing image loader 0x555d98791c10 bytes_read=27509
    0.272889 (+00000.000587) layout_editors_reload_idle_cb: get_desktop_files
Couldn't load icon ufraw: Icon 'ufraw' not present in theme Papirus
Couldn't load icon ufraw: Icon 'ufraw' not present in theme Papirus
Couldn't load icon ufraw: Icon 'ufraw' not present in theme Papirus
Couldn't load icon /usr/share/pixmaps/meshlab.png: Icon '/usr/share/pixmaps/meshlab.png' not present in theme Papirus
Couldn't load icon /usr/share/pixmaps/meshlab: Icon '/usr/share/pixmaps/meshlab' not present in theme Papirus
    0.295869 (+00000.022980) read ahead done for :/home/jwhendy/grab_2020-04-12_123636.png
    0.295888 (+00000.000019) image load completed "/home/jwhendy/grab_2020-04-12_123636.png" (preload)
freeing image loader 0x555d992d42b0 bytes_read=701526
    0.298044 (+00000.002156) layout_editors_reload_idle_cb: setup_editors
    0.298895 (+00000.000851) layout_editors_reload_idle_cb: setup_editors done
    0.318505 (+00000.019610) image load completed "/home/jwhendy/test.png" (current)
    0.318527 (+00000.000022) pixbuf renderer done 0x555d98ea0210
image activate focus_in 0
monitor /home/jwhendy
monitor /home/jwhendy/test.png
monitor /home/jwhendy
monitor /home/jwhendy/test.png
client count 1
client count 2
HUP detected, closing client.
client count 1
client count 2
HUP detected, closing client.
client count 1

###### output stopped here and I force killed
^C
```

### ISSUE TYPE
* Bug Report

### GEEQIE VERSION
```
$ geeqie --version
Geeqie 1.6+git20210208-5901ff85 GTK3
```
Compiled via the AUR pacakge [geeqie-git](https://aur.archlinux.org/packages/geeqie-git/) via [this PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=geeqie-git)
```
$ pacman -Q |grep gtk
clutter-gtk 1.8.4-2
colord-gtk 0.2.0-2
gtk2 2.24.33-1
gtk3 1:3.24.24-2
gtkmm3 3.24.3-1
gtkspell3 3.0.10-2
libfm-gtk2 1.3.1-2
wxgtk-common 3.0.5.1-2
wxgtk3 3.0.5.1-2
```

### OS / DISTRIBUTION
Arch linux
```
$ uname -a
Linux 5.10.14-arch1-1 #1 SMP PREEMPT Sun, 07 Feb 2021 22:42:17 +0000 x86_64 GNU/Linux
```

Happy to provide other output upon request.

**Comment by caclark** [comment]
Could you set up two folders, each with a single image file.
Open two terminal windows and from each run geeqie on one or other of the images with the additional option of `--debug=4`.
Also before running, check that file `$HOME/.config/geeqie/.command` does not exist.

**Comment by jwhendy** [comment]
This appears to be gone. I asked on that arch bug if my observation warranted a re-opening and a maintainer said that it would be fixed with another fix, which came later that day via `2.66.7-1`.

I'll close, re-opening and debugging if this comes back. Thanks for the quick assistance!
